src/format/parse.rs

```diff
@@ -292,13 +296,37 @@ where
                 s = &s[prefix.len()..];
             }
 
-            Item::Space(_) => {
-                s = s.trim_left();
+            Item::Space(item_space) => {
```

Comment:
The implementations here now look very similar to the `Item::*Literal` ones - what are your thoughts on the following code? We could perhaps even deprecate `Item::*Space` and just always use literal? Maybe we need a different item to allow for optional whitespace/literals.

```rust
            Item::Literal(prefix) | Item::Space(prefix) => {
                if s.len() < prefix.len() {
                    return Err((s, TOO_SHORT));
                }
                if !s.starts_with(prefix) {
                    return Err((s, INVALID));
                }
                s = &s[prefix.len()..];
            }

            #[cfg(any(feature = "alloc", feature = "std", test))]
            Item::OwnedLiteral(ref prefix) | Item::OwnedSpace(ref prefix) => {
                if s.len() < prefix.len() {
                    return Err((s, TOO_SHORT));
                }
                if !s.starts_with(&prefix[..]) {
                    return Err((s, INVALID));
                }
                s = &s[prefix.len()..];
            }
```

Reply:
I noticed the same. Seems like a good change to me. However, I decided that this should be put into a different PR so this PR is not overloaded.
If you'd like, I could open a _Code Improvement_ Issue for this.

Reply:
Excellent, I like the idea of this being in a separate PR